tests/std/tests/GH_001017_discrete_distribution_out_of_range/bad_random_engine.hpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+#include <iterator>
+#include <limits>
+#include <type_traits>
+
+namespace detail {
+    struct bad_rng_pattern_sentinel {};
+
+    template <typename UInt, int Width = std::numeric_limits<UInt>::digits>
+    class bad_rng_pattern_generator { // generates bit patterns for bad_random_engine
+    public:
+        using difference_type   = std::ptrdiff_t;
+        using value_type        = UInt;
+        using pointer           = const UInt*;
+        using reference         = const UInt&;
+        using iterator_category = std::input_iterator_tag;
+        using my_iter           = bad_rng_pattern_generator;
+        using my_sentinel       = bad_rng_pattern_sentinel;
+
+        static constexpr value_type top_bit    = value_type{1} << (Width - 1);
+        static constexpr value_type lower_bits = top_bit - 1;
+        static constexpr value_type mask_bits  = top_bit | lower_bits;
+        static constexpr int final_bit_count   = (Width - 1) / 2 + 1;
+
+        constexpr reference operator*() const noexcept { // gets the current pattern
+            return current_value_;
+        }
+
+        constexpr pointer operator->() const noexcept {
+            return &current_value_;
+        }
+
+        constexpr my_iter& operator++() noexcept { // generates the next pattern
+            current_value_ = (current_value_ & lower_bits) << 1 | (current_value_ & top_bit) >> (Width - 1);
+
+            if (current_shift_ < Width - 1 && current_bit_count_ != 0 && current_bit_count_ != Width) {
+                ++current_shift_;
+                return *this;
+            }
+
+            current_shift_ = 0;
+
+            if (current_bit_count_ < final_bit_count) { // n 1's -> n 0's
+                current_bit_count_ = Width - current_bit_count_;
+                current_value_ ^= mask_bits;
+            } else if (current_bit_count_ > final_bit_count) { // n 0's -> (n+1) 1's
+                current_bit_count_ = Width - current_bit_count_ + 1;
+                current_value_     = (current_value_ ^ mask_bits) << 1 | value_type{1};
+            } else { // all bit patterns have been generated, back to all 0's
+                current_bit_count_ = 0;
+                current_value_     = value_type{0};
+            }
+
+            return *this;
+        }
+
+        constexpr my_iter operator++(int) noexcept {
+            const my_iter old = *this;
+            ++*this;
+            return old;
+        }
+
+        friend constexpr bool operator==(const my_iter& a, const my_iter& b) noexcept {
+            return *a == *b;
+        }
+
+        friend constexpr bool operator!=(const my_iter& a, const my_iter& b) noexcept {
+            return !(a == b);
+        }
+
+        friend constexpr bool operator==(const my_iter& iter, const my_sentinel sentinel) noexcept {
+            (void) sentinel;

```

Comment:
While this is correct, it's simpler to use an unnamed parameter, and then our conventions don't need `const` (can't modify something with no name :smile_cat: ):
```suggestion
        friend constexpr bool operator==(const my_iter& iter, my_sentinel) noexcept {
```